crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -1,445 +1,84 @@
-                .get_and_set_internal_id(self.end_point.device_id.clone())
-                .await
-            {
-                error!(
-                    "An error occurred while retrieving internal Id, operation will retry in {} seconds\n Error: {:?}",
-                    RETRY_TIMEOUT_SECS, error
-                );
-
-                match tokio::time::timeout(
-                    Duration::from_secs(RETRY_TIMEOUT_SECS),
-                    self.peers.clients.recv_signal(),
-                )
-                .await
-                {
-                    Ok(Some(RuntimeRequest::Shutdown)) => {
-                        // Give up as requested
-                        return Err(C8YConnectionError::Interrupted);
-                    }
-                    Err(_timeout) => {
-                        // No interruption raised, so just continue
-                        continue;
-                    }
-                    Ok(None) => {
-                        // This actor is not connected to the runtime and will never be interrupted
-                        tokio::time::sleep(Duration::from_secs(RETRY_TIMEOUT_SECS)).await;
-                        continue;
-                    }
-                }
-            };
-        }
-        info!(target: self.name(), "initialisation done.");
-        Ok(())
-    }
-
-    async fn get_and_set_internal_id(&mut self, device_id: String) -> Result<(), C8YRestError> {
-        let internal_id = self.try_get_internal_id(device_id.clone()).await?;
-        self.end_point.set_internal_id(device_id, internal_id);
-
-        Ok(())
+    pub fn new(config: C8YHttpConfig, http: ClientMessageBox<HttpRequest, HttpResult>) -> Self {
+        let end_point =
+            C8yEndPoint::new(&config.c8y_http_host, &config.c8y_mqtt_host, config.proxy);
+        C8YHttpProxyActor { end_point, http }
     }
 
-    pub(crate) async fn try_get_internal_id(
-        &mut self,
-        device_id: String,
-    ) -> Result<String, C8YRestError> {
-        let url_get_id: String = self.end_point.get_url_for_internal_id(&device_id);
-        self.refresh_headers().await?;
-
-        let mut attempt = 0;
-        let mut token_refreshed = false;
-        loop {
-            attempt += 1;
-            let request = HttpRequestBuilder::get(&url_get_id)
-                .headers(&self.end_point.headers)
-                .build()?;
-            let endpoint = request.uri().path().to_owned();
-            let method = request.method().to_owned();
-
-            match self.peers.http.await_response(request).await? {
-                Ok(response) => {
-                    match response.status() {
-                        StatusCode::OK => {
-                            let internal_id_response: InternalIdResponse = response.json().await?;
-                            let internal_id = internal_id_response.id();
+    pub async fn try_get_internal_id(&mut self, device_id: &str) -> Result<String, C8YRestError> {
```

Comment:
I'm assuming that even the remaining functions in this "non-actor" would be moved to `handle` in the next step.